test/test-core.js

```diff
@@ -247,6 +247,180 @@
 		});
 	});
 
+	t('Dynamic response status callback - list of statuses as an array (errors only)', function(assert) {
+		var done = assert.async();
+		var possibleStatuses = [500,400,404];
+	  	var returnedStatuses = [];
+		var maxNumLoops = possibleStatuses.length * 50;
+		var numLoops = 0;
+		var numLoopsComplete = 0;
+
+		$.mockjax({
+			url: '/response-callback',
+			response: function() {
+				this.status = possibleStatuses;
+				this.statusText = 'Internal Server Error';
+			}
+		});
+		
+		var completeCallback = function(xhr) {
+			assert.notEqual($.inArray(xhr.status, possibleStatuses), -1, 'Dynamically set random response status found');
+
+			if( $.fn.jquery !== '1.5.2') {
+				// This assertion fails in 1.5.2 due to this bug: http://bugs.jquery.com/ticket/9854
+				// The statusText is being modified internally by jQuery in 1.5.2
+				assert.equal(xhr.statusText, 'Internal Server Error', 'Dynamically set response statusText matches');
+			}
+
+			// add this to our array of returned statuses (if it isn't there already)
+			if($.inArray(xhr.status, returnedStatuses) === -1) {
+				returnedStatuses.push(xhr.status);
+			}
+
+			// increment counter
+			numLoopsComplete++;
+
+			// if we made it this far without matching all possible statuses, fail!
+			if (numLoopsComplete >= maxNumLoops) {
```

Comment:
This is good, but there's no reason to go through all 150 iterations if we've already found the 3 statuses we need. Can we just add `if (... || returnedStatuses.length === possibleStatuses.length) {` to this conditional?

Reply:
Looks like we didn't capture this change. As it stands, this test will run 300 (mocked) Ajax calls, even if all status codes are returned in the first 10 Ajax calls. If we switch the Ajax call to synchronous, and add the if condition in previous comment I think we can short circuit this and make the test a lot more efficient.